### DISCUSSION TITLE
Add central boiler control feature

### DISCUSSION THREAD

**Discussion by jmcollin78** [jmcollin78/versatile_thermostat#339] [Announcements]
This beta release is released to allow some test in real environment that I cannot do myself because I don't have any central boiler in my house.

This release have been tested in local environment with a fake boiler. The objective of the pre-release is to see in real conditions if its works as expected.

If you want to test it, you are welcome. Please report your experience in the discussion attached to this pre-released.

<hr /><em>This discussion was created from the release <a href='https://github.com/jmcollin78/versatile_thermostat/releases/tag/5.3.3.beta1'>Add central boiler control feature</a>.</em>

**Comment by jmcollin78** [comment]
Hello @HACS-bank, @garywg , @johnbur1 , @pieri70,

This is the main place to comment, add suggestions and share your experience with other users of the new central boiler feature of Versatile Thermostat.

The documentation have been updated here (not on the main branch now): [EN](https://github.com/jmcollin78/versatile_thermostat/tree/5.3.3.beta1?tab=readme-ov-file#controlling-a-central-boiler), [FR](https://github.com/jmcollin78/versatile_thermostat/blob/5.3.3.beta1/README-fr.md#le-contr%C3%B4le-dune-chaudi%C3%A8re-centrale)

Please do the comments in English so that everybody will be able to contribute. (Google translate is your friend if needed).

**Comment by johnbur1** [comment]
Hi @jmcollin78 many thanks for this!
![Screenshot 2024-01-15 10 46 12](https://github.com/jmcollin78/versatile_thermostat/assets/117590638/e78bee1e-3216-4c2e-9a7f-be21436e68a1)
After ticking 'add central boiler' I get this.
What should go in these options?

**Reply by jmcollin78** [reply]
You should refresh your brother to have the new labels. Option + Reload on Mac. Ctrl+F5 (If i remember well) on Windows.
Or lean your brother cache.

**Reply by MBRetrofit** [reply]
I did everything, refresh, delete cache, Chrome, Edge, Android HA app, Vtherm V5.4.1...still the same empty pop out.

**Reply by jmcollin78** [reply]
I have same problem with LocalTuya and Tuya. All field labels are empty. There is something really weird in HA. I think this is independant of VTherm.

![Capture d’écran 2024-02-04 à 20 52 46](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/f212c229-d7c2-45c4-9a4f-8f96f0096efe)

**Comment by pieri70** [comment]
I'll take a look as soon as I have some time, thanks!

**Comment by brunoleon** [comment]
Hello,

nice feature I did exactly that using groups up until now and it is working perfectly so I guess this will too.
I'll try it.

Thanks

**Comment by johnbur1** [comment]
@jmcollin78 have had this working well for two days with a gas boiler switched with a Shelly. Working well - many thanks!
The only suggestion I have is maybe to have the option of how many vtherms are calling for heat before the boiler is fired.
I have 12 vtherms, so waiting for two or three to demand heat before firing the boiler can avoid long running periods with little effect to comfort, and also option to turn off boiler when only one vtherm calling.
This is what my own automation did before enabling yours, and I am seeing long boiler running times now (although it is currently sub zero in Uk, so not typical at the moment).

**Reply by garywg** [reply]
will be downloading this weekend to start testing - thanks for this update.

I too had logic had a 'threshold' of TRVs that must be calling for heat in my old setup, once the number of TRVs was equal or great than the threshold only then will the boiler come on.   Otherwise boiler runs 99% of time.     Busy working on reworking that logic in prep for this update.

On my dashboard have this card setup so I can see each TRV history when calling for heat, a merge of all of these called 'heat required' and finally the boiler operation as well.     By just setting the threshold to 2, something like 30%-45% reduction in boiler runtime. Threshold of  3 was 35% - 55% reduction in boiler runtime so could be very beneficial feature could be done and still keeping things simple.


![sample](https://github.com/jmcollin78/versatile_thermostat/assets/116686494/8f6cc29b-513e-4549-b97e-fe5a7e4f1f54)

**Reply by jmcollin78** [reply]
Interesting comment yes. I will do that. Seems very easy and more "tunable"

**Reply by garywg** [reply]
It was a real eye opener to see the boiler running 99%, and then the signifcant reduction using the threshold - I've got the threshold set as a variable that I can change on a card in my case, but even one set in the config would be hugely useful.  

Normally run threshold at 3 but in my opinion this will vary for everyone based on house, rooms, heat retention for each room, windows, etc etc.   but real easy to achieve tuning.   Never tested above threshold of 5 as could see from each room temp history that first rooms to request heat were getting more than 1 degree below target temp in my case when at 5.  In my case some rooms have doors mostly closed, in house where doors always open and heat can 'escape' to other rooms, maybe higher threshold can achieve even more reduction in boiler runtime for not very much variation from the target temp.

**Reply by jmcollin78** [reply]
> I've got the threshold set as a variable

I guess the variable is an input_number. Right ? What I could is to create a number sensor which holds the threshold. Then you can directly tune by changing this value directly in the entities of the VTherm device. The default value could be 1, so that the behavior is the default behavior explained above. And this avoid adding more configuration parameter which are already too many.

What do you think about ?

**Reply by garywg** [reply]
Hi, yes it is set as an input number which is the threshold.  And then I use a sensor to count the actual TRVs requesting heat:

```
      # Count the number of TRVs that are requesting heat
      - name: "Count of TRVs Requesting Heat"
        unique_id: count_of_trvs_requesting_heat
        icon: mdi:counter
        state: "{{
        expand('group.trvs_requesting_heat') | 
        selectattr('state', 'eq', 'on') | list | count | int 
        }}"
```

and then in the automation that will turn on the boiler (value template below), I do the compare and trigger or not the boiler;

 ```
 action:
    - if:
        - condition: state
          entity_id: binary_sensor.someone_home
          state: "on"
        - condition: template
          value_template: >-
            {{
            (states('sensor.count_of_trvs_requesting_heat') | float) >=
            (states('input_number.trvs_requesting_heat_threshold') | float) 
            }}
      then:
***I trigger the boiler here ***
```

I also use that sensor count on a card;
![Screenshot 2024-01-19 at 11 22 48](https://github.com/jmcollin78/versatile_thermostat/assets/116686494/90e8bc4e-da55-49cc-9af4-8eb0d38f8f2c)

For the default I have mine set to 2 using initial value but certainly for me the only time this was a problem is when restarting HASS which is not very often, I set to 2 on the basis "If forget to manually adjust after a restart, I don't want boiler running 99% of time and maybe not notice for some time = $$$".   But again I think this is likely different depending on each house and each person will have different view on what is "their personal default".   And therefore, Yes, agree - go with option to avoid too many config parameters to keep simple.

```
input_number:
# Max is set to the total number of TRVS you have installed
  trvs_requesting_heat_threshold:
    name: TRVs Requesting Heat Threshold
    initial: 2
    min: 1
    max: 11
    step: 1
    mode: box
```

**Reply by jmcollin78** [reply]
This could be another sensor if "central boiler" feature is selected.

**Comment by jmcollin78** [comment]
@garywg , @pieri70, @johnbur1, @HACS-bank, 
here is the beta release with the enhancement suggested by @garywg just above.

Thanks, if you can have a test in the real environment.

https://github.com/jmcollin78/versatile_thermostat/releases/tag/5.3.3.beta4

---
As proposed by [@garywg](https://github.com/garywg) here https://github.com/jmcollin78/versatile_thermostat/discussions/339, I add two entities to control the boiler starts:

an entity of type sensor counting in real time the number of devices which are heating (and that is marked as "controls the boiler"). If a VTherm have many underlyings, each underlying counts for one. I don't count the VTherm but the number of underlyings entities which are heating,
an entity of type number which holds the threshold of devices number heating under which we don't start the boiler.
It is working on my test environment but I need tests in real condition to be sure.

**Comment by johnbur1** [comment]
Hi @jmcollin78 this is fantastic - many thanks for all your work on this.

Can I suggest one more enhancement for gas boiler users? My initial testing is finding a lot of 'short cycling' - the boiler turning on/off too often.
Would it be possible to add other sensors that control how long the boiler runs for after firing - eg run for a minimum of X minutes - would need a check that at least one vtherm is still calling for heat (any vtherm, not just ones centrally controlled).
Conversely, another sensor to control how long the boiler stays off before firing again.
Hope this makes sense. 

![share_7935695067381051972](https://github.com/jmcollin78/versatile_thermostat/assets/117590638/d72f9950-bd17-4c13-9a89-4bd2d3b36b95)

**Reply by jmcollin78** [reply]
Thank you for your report !

Do you have seen the threshold I add for this purpose ? (the number of heaters devices that should be on before turning on the boiler). 

If I add the parameters you speak about, the could be cases in which the boiler is activated and no heaters are on. I understand we really don't want this. This could damage the pump of the boiler (If I understand well - I don't have any in my home).

**Reply by johnbur1** [reply]
My scenario: I have 15 trvs groups into 13 vtherms (2 rooms have 2 radiators). I have marked 10 of the vtherms to be controlled by the central boiler (the other 3 are less used rooms, so I am happy they only get heat when other vtherms have turned the boiler on. I currently have the threshold set to 3 to turn the boiler on, but i am finding short cycling as reported above.
Currently:
3 or more vtherms call for heat, so boiler fires.
Shortly after, this reduces to 2 vtherms, so boiler stops (plus some of the vtherms not controlled by the central boiler could still be open available to take heat).
Shortly after this, a third vtherm calls for heat again so boiler fires again.
etc.. etc..

My thought was to have a user defined timer for minimum firing time to avoid this situation (and corresponding timer to stop the boiler firing again too quickly).

You are correct, as I said above, there would need to be a check that at least one vtherm is still calling for heat (any vtherm, not just ones centrally controlled).
Most modern systems (in UK at least?) have a by-pass valve to avoid damage should the boiler run with all trvs closed, but certainly a waste of gas.

An alternative solution could be to have one parameter to turn the boiler on (eg set to 3 vtherms), and another to turn the boiler off (eg set to 1 vtherm)?

**Reply by jmcollin78** [reply]
What about setting the threshold to 2 and/or have longer cycle and/or change the minimum activation duration on each TRV (in % for `over_valve` VTherm) ? . This should drastically reduce the short activation number.

There is no fully satisfying solution because we face contradictory injunctions:
1. avoid short activation,
2. guarantee than the boiler will never be on and all TRV off. This seems to be a real issue for other people (I've seen comments that this could break the pump). 
3. minimize the ernergy consumption

So there is a compromise to find. Having a tunable threshold + tunable cycle duration + tunable minimal % of activation  allows each user to have each own optimization criteria. So, for now, I do not intend to do more and complexify the tuning. All is in your hands 😉  (unless many people have a common problem).

**Reply by johnbur1** [reply]
Thank you, fair enough, I will try all options.
Re: 2. There must be a lot of very old pipework still out there....😉😉

**Reply by jmcollin78** [reply]
> Re: 2. There must be a lot of very old pipework still out there...

Yes. I was discussing yesterday with someone that explains me that low cost central boiler or PAC don't have some security. This can lead to damage. I have added a disclaimer in the readme for this. I don't want that for a bug or something like that, someone have damage its central boiler.

**Reply by pieri70** [reply]
Usually at least one radiator should be always open, without TRV. The bathroom one for example. This way if the boiler starts and pumps water there is a minimal flow

**Comment by pieri70** [comment]
Hi
I finally have some time to test this feature, sorry for delay
In my case I don't have a real switch dedicated to turning on/off my boiler, instead I have my old tuya wifi thermostat, connected to the boiler, on which I set target temperature (high/low) if I need/don't need heating. So the thermostat works as a "switch", it does not control the temperature itself..
I use an automation with two id (15=low target temperature, 25=high target temperature) and I set high target temperature when two or more trv request heat, I turn off boiler (low target temperature) when all trv go in idle state
I use two cases >1 then target =25°C and == 0 then target =15°C because I like my boiler to turn on when there is a real need of heating (two or more rooms) and I turn it off when all rooms are heated enough - there is always a room that has bigger inertia and takes longer to warm up

```
alias: turn_on_boiler
description: Turn on my boiler if >1 trv is requesting heat
trigger:
  - id: "25"
    platform: template
    value_template: >
      {{ ['climate.trv_1', 'climate.trv_2',
      'climate.trv_3','climate.trv_4','climate.trv_5'] |
      select('is_state_attr', 'hvac_action', 'heating') | list | count > 1 }}
  - id: "15"
    platform: template
    value_template: >
      {{ ['climate.trv_1', 'climate.trv_2',
      'climate.trv_3','climate.trv_4','climate.trv_5'] |
      select('is_state_attr', 'hvac_action', 'heating') | list | count == 0 }}
condition: []
action:
  - service: climate.set_temperature
    target:
      entity_id: climate.home
    data:
      temperature: "{{ trigger.id|int(15) }}"
```

To use your function how can I convert the needed input
entity_id/service_id[/attribute:value]
into a command that works for my case?
climate.home/climate.set_temperature/temperature:25 to turn on and
climate.home/climate.set_temperature/temperature:15 to turn off?

Or is there some way to create a "virtual switch" that controls these values?

Sorry for basic question but I'm a rookie ;)

Thanks
Pietro

**Reply by jmcollin78** [reply]
Yes you can use directly the service configuration like:

```
climate.home/climate.set_temperature/temperature:25 to turn on and
climate.home/climate.set_temperature/temperature:15 to turn off
```
It have been done exactly for this :wink:

**Comment by johnbur1** [comment]
https://github.com/jmcollin78/versatile_thermostat/discussions/339#discussioncomment-8212972
Hi @jmcollin78 do you think that you will do something with this, or should I do my own solution?
Many thanks for all your great work

**Reply by jmcollin78** [reply]
answered. Normally I answer all comments (when I have the answer of course). So no need to kindly remind me. Thank you (I have a lot of notification coming from many sources to deal with).

**Reply by johnbur1** [reply]
Indeed! Understand ;-)

**Comment by MBRetrofit** [comment]
This can be useful for over_switch too when something like when all Vtherm is on idle or Off the electric boiler could start ( if needed).

I already made a binary_sensor based on a simple template ( but this is not exactly in real time as depends more on HA timing)

`{% set hot = states.climate
|selectattr('attributes.hvac_action', 'eq', 'heating')
|map(attribute='entity_id') |list |count |int(0) %}
{{ hot > 0 }}`

I would prefer to use your switch for obtaining something like:

![Boiler PID](https://github.com/jmcollin78/versatile_thermostat/assets/151366453/82c35e0f-1994-4335-b6fb-ba41f17849a5)

This way the minimum power load on house wires can be maintained and more important the heating element of the boiler does not heat cold water as when time pass the water temp inside the boiler will raise naturally in the bottom part.

**Reply by jmcollin78** [reply]
This works also for `over_switch` VTherm. An `over_switch` VTherm can control the boiler (the checkbox is present too).

**Comment by MBRetrofit** [comment]
Indeed, but because I only saw empty checkbox even after clearing the browser cache, I did not understand if the boiler is with `condition: not` as in my full electric heating system I will need boiler to be On when all the Vtherm's is on Idle or Off.
Now is it like this:

![Boiler](https://github.com/jmcollin78/versatile_thermostat/assets/151366453/b8eb06b3-e934-4ba0-940d-f448d3808180)


and my Chrome shows:

![Config](https://github.com/jmcollin78/versatile_thermostat/assets/151366453/d05169c2-c9b7-4211-a012-22632248da09)

**Reply by jmcollin78** [reply]
Clear the cache and re-clear the cache. For download CTRL+F5. Sometimes, dont know why, it don't work the first time.

**Reply by MBRetrofit** [reply]
I did all that, nothing changed

**Reply by jmcollin78** [reply]
What is you installed release ?

**Reply by MBRetrofit** [reply]
V5.3.3

**Comment by Banzarykey** [comment]
Loving this feature but is it possible to use hvac_action: heating rather than state: heating? I use auto climate on most of my TRV's and they don't currently trigger the central boiler control :/

**Comment by jmcollin78** [comment]
Hello @Banzarykey ,

Depending of your VTherm type the nb of active device is calculated differently:
1. `over_switch`: the state of the switch which control the heater should be 'on'. For a switch there is no hvac_action,
2. `over_valve`: the open percent of the valve should be > 0%. Their is also no hvac_action possible for this,
3. `over_climate`: the hvac_mode of the underlying should be not 'OFF' and the hvac_action should be not in IDLE or OFF.

So I guess what you are requesting is already in place or not possible.

**Reply by Banzarykey** [reply]
Thanks @jmcollin78 

Thats interesting im using over_climate below is the list of attributes when the TRV is set to auto 
![image](https://github.com/jmcollin78/versatile_thermostat/assets/30878682/f91d6f8f-13e8-4a80-a5fe-593b5c6ae7e7)

from my understanding in that state the binary_sensor.central_boiler state should be on? 
![image](https://github.com/jmcollin78/versatile_thermostat/assets/30878682/e77b83af-d2c0-4fec-8600-2d76aeb08f33)

and the config of VT
![image](https://github.com/jmcollin78/versatile_thermostat/assets/30878682/bc292861-5195-4614-942a-39ed7d24cb90)

maybe a bug?

**Comment by jmcollin78** [comment]
> Thats interesting im using over_climate below is the list of attributes when the TRV is set to auto
> ![image](https://private-user-images.githubusercontent.com/30878682/307316999-f91d6f8f-13e8-4a80-a5fe-593b5c6ae7e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg4NTg4MjYsIm5iZiI6MTcwODg1ODUyNiwicGF0aCI6Ii8zMDg3ODY4Mi8zMDczMTY5OTktZjkxZDZmOGYtMTNlOC00YTgwLWE1ZmUtNTkzYjVjNmFlN2U3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI1VDEwNTUyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5NjJlY2E1NjQxYTM1MDU1ODQxZmJmOGZmNmUyMjM1MTEzNTRlYmY1YjAyNDIwMzYyMWU4MWQxYjgwNzk1N2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HRSquL273EWc3thxbn-z-cmsUOi-asixrAobo2ByeJw)

The relevant code is the following:

```
@property
    def is_device_active(self):
        """If the toggleable device is currently active."""
        if self.is_initialized:
            return (
                self._underlying_climate.hvac_mode != HVACMode.OFF
                and self._underlying_climate.hvac_action
                not in [
                    HVACAction.IDLE,
                    HVACAction.OFF,
                ]
            )
        else:
            return None
```
So with hvac_mode = 'auto' and hvac_action='heating' it should work (and it is working on my env).

Can you try, setting the hvac_mode to Heat and not to Auto ? It should not change anything for you.

Other thing to check: check that the `is_device_active` is True in the VTherm attributes when you have the bug.

**Reply by Banzarykey** [reply]
When heat is selected it works as expected, VTherm is set to: is_over_climate: true in both heat and auto modes. 

![image](https://github.com/jmcollin78/versatile_thermostat/assets/30878682/6dbbf87b-dfc2-4117-ad95-f9a11bfd28a0)
![image](https://github.com/jmcollin78/versatile_thermostat/assets/30878682/0d2d18f0-d2fa-419f-85fb-456e2c0bf259)

Debug of heat mode
`2024-02-26 08:48:57.099 DEBUG (MainThread) [custom_components.versatile_thermostat.binary_sensor] VersatileThermostat-Central boiler - calculating the new central boiler state
2024-02-26 08:48:57.099 INFO (MainThread) [custom_components.versatile_thermostat.binary_sensor] VersatileThermostat-Central boiler - central boiler have been turned on
2024-02-26 08:48:57.099 INFO (MainThread) [custom_components.versatile_thermostat.const] VersatileThermostat-Central boiler - Sending event EventType.CENTRAL_BOILER_EVENT with data: {'central_boiler': True}
2024-02-26 08:48:57.684 DEBUG (MainThread) [custom_components.versatile_thermostat.thermostat_climate] VersatileThermostat-Kitchen - _async_climate_changed new_state is <state climate.kitchen_trv=heat; hvac_modes=['auto', 'heat', 'off'], min_temp=5.0, max_temp=35.0, target_temp_step=0.5, preset_modes=['none', 'auto', 'manual', 'off', 'on'], current_temperature=13.3, temperature=28.5, hvac_action=heating, preset_mode=on, friendly_name=Kitchen TRV, supported_features=401 @ 2024-02-26T08:48:57.096628+00:00>
2024-02-26 08:48:57.685 INFO (MainThread) [custom_components.versatile_thermostat.thermostat_climate] VersatileThermostat-Kitchen - Underlying climate climate.kitchen_trv changed. Event.new_hvac_mode is heat, current_hvac_mode=heat, new_hvac_action=heating, old_hvac_action=heating
2024-02-26 08:48:57.685 DEBUG (MainThread) [custom_components.versatile_thermostat.thermostat_climate] VersatileThermostat-Kitchen - last_change_time=2024-02-26 08:48:57.075466+00:00 old_state_date_changed=2024-02-26 08:48:57.096628+00:00 old_state_date_updated=2024-02-26 08:48:57.096628+00:00 new_state_date_changed=2024-02-26 08:48:57.096628+00:00 new_state_date_updated=2024-02-26 08:48:57.684410+00:00
2024-02-26 08:48:57.685 INFO (MainThread) [custom_components.versatile_thermostat.thermostat_climate] VersatileThermostat-Kitchen - underlying event is received less than 10 sec after command. Forget it to avoid loop
2024-02-26 08:48:57.685 DEBUG (MainThread) [custom_components.versatile_thermostat.sensor] VersatileThermostat-Nb device active for boiler - calculating the number of active VTherm
2024-02-26 08:48:57.685 DEBUG (MainThread) [custom_components.versatile_thermostat.sensor] Examining the hvac_action of Office
2024-02-26 08:48:57.685 DEBUG (MainThread) [custom_components.versatile_thermostat.sensor] Examining the hvac_action of Living Room
2024-02-26 08:48:57.685 DEBUG (MainThread) [custom_components.versatile_thermostat.sensor] Examining the hvac_action of Master
2024-02-26 08:48:57.685 DEBUG (MainThread) [custom_components.versatile_thermostat.sensor] Examining the hvac_action of Kitchen`

Debug of auto state:
`2024-02-26 08:48:58.028 DEBUG (MainThread) [custom_components.versatile_thermostat.thermostat_climate] VersatileThermostat-Kitchen - _async_climate_changed new_state is <state climate.kitchen_trv=heat; hvac_modes=['auto', 'heat', 'off'], min_temp=5.0, max_temp=35.0, target_temp_step=0.5, preset_modes=['none', 'auto', 'manual', 'off', 'on'], current_temperature=13.4, temperature=28.5, hvac_action=heating, preset_mode=on, friendly_name=Kitchen TRV, supported_features=401 @ 2024-02-26T08:48:57.096628+00:00>
2024-02-26 08:48:58.028 INFO (MainThread) [custom_components.versatile_thermostat.thermostat_climate] VersatileThermostat-Kitchen - Underlying climate climate.kitchen_trv changed. Event.new_hvac_mode is heat, current_hvac_mode=heat, new_hvac_action=heating, old_hvac_action=heating
2024-02-26 08:48:58.028 DEBUG (MainThread) [custom_components.versatile_thermostat.thermostat_climate] VersatileThermostat-Kitchen - last_change_time=2024-02-26 08:48:57.075466+00:00 old_state_date_changed=2024-02-26 08:48:57.096628+00:00 old_state_date_updated=2024-02-26 08:48:57.684410+00:00 new_state_date_changed=2024-02-26 08:48:57.096628+00:00 new_state_date_updated=2024-02-26 08:48:58.025881+00:00
2024-02-26 08:48:58.028 INFO (MainThread) [custom_components.versatile_thermostat.thermostat_climate] VersatileThermostat-Kitchen - underlying event is received less than 10 sec after command. Forget it to avoid loop
2024-02-26 08:48:58.028 DEBUG (MainThread) [custom_components.versatile_thermostat.sensor] VersatileThermostat-Nb device active for boiler - calculating the number of active VTherm
2024-02-26 08:48:58.028 DEBUG (MainThread) [custom_components.versatile_thermostat.sensor] Examining the hvac_action of Office
2024-02-26 08:48:58.028 DEBUG (MainThread) [custom_components.versatile_thermostat.sensor] Examining the hvac_action of Living Room
2024-02-26 08:48:58.028 DEBUG (MainThread) [custom_components.versatile_thermostat.sensor] Examining the hvac_action of Master
2024-02-26 08:48:58.028 DEBUG (MainThread) [custom_components.versatile_thermostat.sensor] Examining the hvac_action of Kitchen`

**Reply by jmcollin78** [reply]
> When heat is selected it works as expected

That is a good news. I will then transform this discussion into an issue to fix that when underlying is in Auto havc_mode.

When I look to the code, it looks good for me but I have to investigate further.

Thank you for your report.

**Comment by jmcollin78** [comment]
https://github.com/jmcollin78/versatile_thermostat/issues/396

**Comment by tbrasser** [comment]
Was looking for exactly something like this.

I have a few questions, I'm still getting all my radiators fitted with TRV's, however my main central heating furnace/boiler is opentherm controlled (modulated), so depending on the ask it modulates.

For now I sync the setpoint from central to my TRV's, and calculate "average temperature in occupied rooms" and use that as the measured temperature for the central thermostat.

I would love for this system to be smarter, what would be a good way to configure this? Set the highest set active vtherm as setpoint on central one? And then lowest measured one as measured temp on my central thermostat? (It's esphome so full control over it)

And do I put a vtherm on top of my central climate as well?

**Reply by jmcollin78** [reply]
Hello @tbrasser ,

I'm not totally sure of the good understanding. But if your central boiler modulate considering the ask, there is nothing more to do. Let the system modulate itself and just add VTherm over each TRV your have.

If you want to control the target temperature (setpoint) of your central boiler, you can create a central configuration with central boiler activated (see: https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#control-of-a-central-boiler) and configure the service to activate (resp. deactivate) to change the target temperature. There is example on the README on how to do that. But I'm not sure it is necessary if your system modulate itself.

**Reply by tbrasser** [reply]
So my main thermostat that modulates the boiler has no physical sensor. (it used to be a non smart model in the living room.) I've been adding smart TRVs to other rooms and in the future I would also like to change out the living room non-thermostatic (and never closed) valve for a smart trv.

My main question is how to control automatically my central boiler (which is a climate component, and for currently sensed temperature I can use any home assistant entity) based on the demand of the smart TRVs.

Let's say I use vtherm over climate on all my smart TRVs, I want to have my central boiler's 'ask' be proportional to what my TRVs are asking for. But I'm unsure of the logic and unsure if vtherm can support anything other than 'on' or 'off'.

**Reply by garywg** [reply]
Hi,

My first comment, you should not remove the 'never' closed valve in the lounge.     Normally always 1 radiator is 'never closed' to allow if the pump runs for water to circulate otherwise damage to the pump can occur.     Normally, this 'never closed' radiator is just one and is normally in the main bathroom.   

I'm not sure what you mean by 'proportional ask' ?    You can set the demand that at least 1, or 2 or 3.....etc TRVs must be asking for heat before VTherm will activate the boiler.   It depends what you primary driver is....'efficient cost'. or absolute keep temperature as close as possible to desired temp as to how many TRVS you set to request heat before boiler will come on.

In my case the central thermostat has a sensor, but this is ignored for temp monitoring, it is only used to set the temp 1.5 higher than current temp, so that when signal to boiler is sent to boiler to switch on is, the boiler will actually switch on because the target temp is 1.5 higher than current.  Other than that the temp sensor on the main thermostat has no function so to speak. (Which 'equates' to I think your setup on the boiler).

Hope this helps.

**Reply by johnbur1** [reply]
All modern systems should include a bypass valve - you don't need to have a 'never' closed valve unless you have a particularly old system.

**Reply by jmcollin78** [reply]
@tbrasser 

> But I'm unsure of the logic and unsure if vtherm can support anything other than 'on' or 'off'.

Yes for controling boiler, the Vtherm supports all services. You have to configure the entity/service and parameter to start/stop central boiler. There is no more than start / stop. Nothing to update a value i already up.
